engine/access/rpc/backend/backend.go

```diff
@@ -226,14 +225,51 @@ func executionNodesForBlockID(
 	state protocol.State) (flow.IdentityList, error) {
 
 	// lookup the receipts storage with the block ID
-	receipts, err := executionReceipts.ByBlockIDAllExecutionReceipts(blockID)
+	allReceipts, err := executionReceipts.ByBlockIDAllExecutionReceipts(blockID)
 	if err != nil {
 		return nil, fmt.Errorf("failed to retreive execution receipts for block ID %v: %w", blockID, err)
 	}
 
+	// execution result ID to execution receipt map to keep track of receipts by their result id
+	var identicalReceipts = make(map[flow.Identifier][]*flow.ExecutionReceipt)
+
+	// maximum number of matching receipts found so far for any execution result id
+	maxMatchedReceiptCnt := 0
+	// execution result id key for the highest number of matching receipts in the identicalReceipts map
+	var maxMatchedReceiptResultID flow.Identifier
+
+	// find the largest list of receipts which have the same result ID
+	for _, receipt := range allReceipts {
+
+		resultID := receipt.ExecutionResult.ID()
+		receipts, found := identicalReceipts[resultID]
+
+		if !found {
+			// first of its kind
+			receipts = []*flow.ExecutionReceipt{receipt}
+			identicalReceipts[resultID] = receipts
+			// keep looking for at least one more receipt with the same result
+			// since we need at least two execution receipt which have the same result
+			continue
+		}
+
+		// append receipt to the list of all receipts for this resultID
+		receipts = append(receipts, receipt)
+		identicalReceipts[resultID] = receipts
+
+		currentMatchedReceiptCnt := len(receipts)
+		if currentMatchedReceiptCnt > maxMatchedReceiptCnt {
+			maxMatchedReceiptCnt = currentMatchedReceiptCnt
```

Comment:
in the Access node case, unlike the consensus node, we don't want to stop collecting identical receipts as soon as we have found two since we want to collect IDs of all execution nodes to choose from (and not just two)